pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -526,10 +542,11 @@ func (r *reconciler) normalizeLoadBalancerServiceAnnotations(service *corev1.Ser
 // on existing resources.
 func (r *reconciler) finalizeLoadBalancerService(ci *operatorv1.IngressController) (bool, error) {
 	haveLBS, service, err := r.currentLoadBalancerService(ci)
+	isMyLBS := isServiceOwnedByIngressController(service, ci)
 	if err != nil {
 		return false, err
 	}
-	if !haveLBS {
+	if !haveLBS || !isMyLBS {
```

Comment:
We have an unusual situation here.  The `finalizeLoadBalancerService` method, and calls to it, can be deleted at this point because we changed the operator not to use the finalizer, and to remove it if it existed, in OpenShift 4.8.  See the comment on the `deleteLoadBalancerServiceFinalizer` method (which likewise can be deleted).  I can file a BZ and PR to remove this unnecessary code.

Reply:
https://github.com/openshift/cluster-ingress-operator/pull/729

Reply:
Ah yes I saw those comments about it not really being necessary. Thanks for cleaning that up.